.github/workflows/ci-stylelint-polaris.yml

```diff
@@ -0,0 +1,30 @@
+name: '@shopify/stylelint-polaris CI'
+
+on:
+  pull_request:
+    paths:
+      - 'stylelint-polaris/**'
+      - 'polaris-react/src/tokens/token-groups/**'
+  push:
+    branches: main
+    paths:
+      - 'stylelint-polaris/**'
+      - 'polaris-react/src/tokens/token-groups/**'
+
+jobs:
+  test:
+    name: Test with node v${{ matrix.node_version }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node_version: ['14.17.0', '16.13.0']
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node_version: ${{ matrix.node_version }}
+          cache: yarn
+
+      - run: yarn --frozen-lockfile
+      - run: yarn type-check
```

Comment:
The `stylelint-polaris` `tsconfig` defines `emitDeclarationOnly: true`: https://github.com/Shopify/polaris/blob/4b02c9743babe60be5ffb80fc124db171a700ece/stylelint-polaris/tsconfig.json#L5-L12

We might need to add a `"types": "dist/index.d.ts"` key to the `stylelint-polaris` package.json